boards/esp8266-olimex-mod/Kconfig

```diff
@@ -13,4 +13,10 @@ config BOARD_ESP8266_OLIMEX_MOD
     select BOARD_COMMON_ESP8266
     select CPU_MODEL_ESP8266_ESP_12X
 
+    select HAVE_RIOT_CI_BUILD_ESP_SW_TIMER
+
+    select MODULE_ESP_LOG_STARTUP if TEST_KCONFIG && RIOT_CI_BUILD
+    select MODULE_ESP_LOG_TAGGED if TEST_KCONFIG && RIOT_CI_BUILD
+    select MODULE_ESP_QEMU if TEST_KCONFIG && RIOT_CI_BUILD
```

Comment:
Would it be an option to instead have another BOARD that inherits from the original enabling all these extra modules? IMO there should not be dependencies against anything that is not a MODULE, FEATURE (or CPU/CPU_MODEL/etc.)

Reply:
Yes that would be another option, the virtual board idea.  I think that would be a different PR then though.  One more board for the CI.  @gschorcht @leandrolanzieri @fjmolinas If we think this is the way forward I would remove the ESP fixes for this PR.

Reply:
Hm, I already had an approach with two pure virtual board definitions, one for ESP8266 and one for ESP32, which use these optional modules. But this approach was discarded because we tried not to further increase the number of boards and thus the compilation time in Murdock.

Reply:
Regarding the virtual boards, we had a discussion in PR #12971.

Reply:
> Regarding the virtual boards, we had a discussion in PR #12971.

I can't find in the PR the arguments against virtual BOARDs, is there a specific thread? IMO custom dependencies is not great, its easy for those to spiral out of control and for the ordeal to become messy as it was in the past, I would probably have blocked #17314 because of this.

Personally if build is an issue what I would have had is to add to the `externals-boards` test two clones of `esp32` and `esp86` with all those modules enabled. Then you could simply add those external-boards in the required test application. I opened https://github.com/RIOT-OS/RIOT/pull/17525 as a demonstrator for that.